src/fsharp/pars.fsy

```diff
@@ -2347,18 +2349,18 @@ recdFieldDeclList:
 
 /* A field declaration in a record type */
 recdFieldDecl: 
-  | opt_attributes  fieldDecl
-     { let fld = $2 $1 false 
-       let (Field(a,b,c,d,e,f,vis,g)) = fld 
-       if Option.isSome vis then errorR(Error(FSComp.SR.parsRecordFieldsCannotHaveVisibilityDeclarations(),rhs parseState 2))
-       Field(a,b,c,d,e,f,None,g)  }
+  | opt_attributes fieldDecl
+     { let wholeRange = rhs2 parseState 1 2
+       let fld = $2 $1 false wholeRange
+       let (Field (a, b, c, d, e, f, vis, wholeRange)) = fld
+       if Option.isSome vis then errorR (Error (FSComp.SR.parsRecordFieldsCannotHaveVisibilityDeclarations (), rhs parseState 2))
+       Field (a, b, c, d, e, f, None, wholeRange) }
 
 /* Part of a field or val declaration in a record type or object type */
 fieldDecl: 
   | opt_mutable opt_access ident COLON  typ 
-     { let mRhs = rhs2 parseState 3 5 
-       let xmlDoc = grabXmlDoc(parseState,3)
-       (fun attrs stat -> Field(attrs, stat,Some $3,$5,$1,xmlDoc,$2,mRhs)) }
```

Comment:
Same as before, looks like we are going to include attributes range as well since `wholeRange` is being passed from the rules above.
I would check to make sure an actual error range doesn't look bad when the field has attributes on the previous line.

Reply:
The errors usually look OK since in many cases they're not covering the whole range, taking a relevant part like an id instead.